Document changing a pipeline's state (#3818)
tenzir/tenzir: web/docs/user-guides/manage-a-pipeline/README.md

---
sidebar_position: 1
---

# Manage a pipeline

A pipeline can enter many states after you [run it][run-a-pipeline]. The
[app](https://app.tenzir.com/) or [API](/api) allow you to manage the pipeline
lifecycles.

[run-a-pipeline]: ../run-a-pipeline/README.md

The diagram below illustrates the various states, where circles correspond to
states and arrows to state transitions:

![Pipeline States](pipeline-states.excalidraw.svg)

The grey buttons indicate the actions you, as a user, can take to transition
into a different state. The orange arrows are transitions that take place
automatically based on system events.

The states have the following meaning:

- **Created**: the pipeline has just been [deployed][run-a-pipeline].

> - **Running**: the pipeline is actively processing events.

Reviewer: ```suggestion
- **Running**: the pipeline is actively processing data.
```

> - **Completed**: there are no more events to process.

Reviewer: ```suggestion
- **Completed**: there is no more data to process.
```
